pkg/reconciler/taskrun/resources/volume.go

```diff
@@ -29,3 +29,45 @@ func GetPVCVolume(name string) corev1.Volume {
 		},
 	}
 }
+
+// appendNewSecretsVolumes takes a variadic list of secret volumes and a list of volumes, and
+// appends any secret volumes that aren't already present therein to the list. The secret volumes
+// are volumes whose VolumeSource is set to *corev1.SecretVolumeSource with only the SecretName
+// field filled in. Specifically, they must have the following structure as defined by
+// (pkg/apis/resource/v1alpha1/storage.ArtifactBucket).GetSecretsVolumes():
+//
+// corev1.Volume{
+//   Name: fmt.Sprintf("volume-bucket-%s", sec.SecretName),
+//   VolumeSource: corev1.VolumeSource{
+//     Secret: &corev1.SecretVolumeSource{
+//       SecretName: sec.SecretName,
+//     },
+//   },
+// }
+func appendNewSecretsVolumes(vols []corev1.Volume, newVols ...corev1.Volume) []corev1.Volume {
+	for _, newv := range newVols {
+		if newv.Secret == nil {
+			panic("appendNewSecretsVolumes: VolumeSource is not a secret")
```

Comment:
Would definitely prefer not to panic in the controller.  Can we just silently add non-secret volumes? And / or maybe return a `multierror`?

Reply:
Good idea. Adding non-secret volumes, even if that means there are duplicates, is still strictly better than the status quo. That makes this PR de-duplicate only secret volumes with the expected structure, which is targets the problem more specifically.